drivers/iio/adc/madura/adrv9025.c

```
@@ -1530,6 +1530,45 @@ static int adrv9025_register_debugfs(struct iio_dev *indio_dev)
 
 #define ADRV9025_MAX_CLK_NAME 79
 
+static int __adrv9025_of_get_u32(struct device_node *np, const char *propname,
+				 s64 defval, void *out_value, u32 size)
+{
+	u32 tmp;
+	u64 tmp64;
+	int ret;
+
+	if (size == 8) {
+		tmp64 = defval;
+		ret = of_property_read_u64(np, propname, &tmp64);
+	} else {
+		tmp = defval;
+		ret = of_property_read_u32(np, propname, &tmp);
+	}
```

You're ignoring error codes from DT. What if the property is not given? If the property is mandatory then you need to act accordingly. But do we really need AGC to be mandatory (given the fact we have lots of properties for it)?

Another thing this does not cover is min/max values. I'm fairly sure some constrains exist for some of the parameter as this code just allows any value to be given...

---

If there is no DT variable with that name, the macro sets a default value.

In case invalid values are given, the API will throw an error. However, the UG specifies the limits and I will add the corresponding logic.